mpqp/tools/circuit.py

```diff
@@ -159,6 +171,53 @@ def random_instruction(
             return gate_class(control, target)
 
 
+def random_noise(
+    noise_model: Optional[Sequence[type[NoiseModel]]] = None,
+    seed: Optional[int | Generator] = None,
+) -> NoiseModel:
+    """This function creates a noise model.
+    The noise are chosen randomly from the provided list of noise model.
+
+    args:
+        noise_model : List of noise model.
+
+    Returns:
+        A quantum circuit with the specified number of qubits and randomly chosen gates.
+
+    Raises:
+        ValueError: If the number of qubits is too low for the specified gates.
+
+    Examples:
+        >>> random_noise() # doctest: +SKIP
+        Depolarizing(0.37785041428875576)
+
+    """
+    rng = np.random.default_rng(seed)
+
+    if noise_model is None:
+        noise_model = NOISE_MODELS
+
+    noise = rng.choice(np.array(noise_model))
+
+    if issubclass(noise, AmplitudeDamping):
+        prob = rng.uniform(0, 1)
```

Comment:
I'm not in love with this, I guess this is the reason why you asked if we couldn't have a `proba` attribute for all noises